Rails7 upgrade

Mate2xo opened the pull request:
Upgrade from Rails 6.0 to 7.1, with fixes along the way.
* missing translations, and extractions to narrower namespaces
* update LSP config
* improve coverage
* ActiveAdmin had to be upgraded from 2 -> 3 for Rails 7.1; which needed upgrading Ransack from 3 -> 4, that requires whitelisting searchable attributes. Admin filters have been improved and simplified along the way
* bumped other gems that would'nt work with Rails 7.0/7.1

There's a lot of lint, this was done on separate commits to ease the review. It might be easier to review this PR commit by commit

Mate2xo committed:
refactor: delete unused methods

Mate2xo committed:
refactor: remove useless inheritance

Mate2xo committed:
test: improve coverage

Mate2xo committed:
chore: lint

Mate2xo committed:
docs: TODO

Mate2xo committed:
feat(Rails upgrade): apply 6.0 defaults

Mate2xo committed:
feat(Rails upgrade): remove direct dependency to mini-magick

https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#active-storage-now-requires-image-processing

Mate2xo committed:
fix: typo

Mate2xo committed:
feat(Rail upgrade): migrate gem to 6.1, adapt rspec fixtures accordingly

Mate2xo committed:
chore: lint

Mate2xo committed:
feat(Rails upgrade): apply 6.1 defaults

Mate2xo committed:
build(Rails upgrade): bump web-console for Rails 6 & 7 support

Mate2xo committed:
config(Rails upgrade): Zeitwerk is now the only option

Mate2xo committed:
config(Rails upgrade): update deprecated config namespace

Mate2xo committed:
build(Rails upgrade): bump Activeadmin for Rails 7 support

Mate2xo committed:
build(Rails upgrade): bump active_storage_validations for Rails 7.1 support

Mate2xo committed:
build(Rails upgrade): bump enumerize for Rails 6.1+ & Ruby 3

Mate2xo committed:
feat(Rails upgrade): migrate to 7.0

Mate2xo committed:
feat(Rails upgrade): apply 7.0 defaults

Mate2xo committed:
fix(concurrent-ruby): avoid logger error, try to remove me with Rails 7.1

Mate2xo committed:
config: variabilize default host

Mate2xo committed:
refactor(model locales): extract to own model-related files

Mate2xo committed:
chore: lint

Mate2xo committed:
fix(layout): show category input in panel instead of 'nowhere'

Mate2xo committed:
chore: lint

Mate2xo committed:
feat(locales): translate all created/updated_at by default

Mate2xo committed:
fix(locales): add missing translations

Mate2xo committed:
chore(lint): alphabetical order

Mate2xo committed:
fix(locales): add missing, avoid raw concatenation

Mate2xo committed:
test: improve test descriptions

Mate2xo committed:
chore: lint

Mate2xo committed:
fix(locales): add missing translations

Mate2xo committed:
config(rubocop): ignore blocklength cop for admin files

Mate2xo committed:
perf(queries): remove N+1, remove unnecessary eager loading

Mate2xo committed:
style: `menu` declarations at the top

Mate2xo committed:
feat(ActiveAdmin upgrade): 2->3, ransack 3->4, fix filters consistency

All ransack attributes (mostly used in AA) must now be whitelisted

Mate2xo committed:
docs(gem): reminder for future upgrades

Mate2xo committed:
docs: TODO

Mate2xo committed:
feat(admin): show week types as tags

Mate2xo committed:
chore: lint

Mate2xo committed:
build(pundit): bump to 2.5

Mate2xo committed:
config(rubocop): bundle extensions

Mate2xo committed:
config(solagraph): bundle extensions

Mate2xo committed:
config(rubocop): using 'Include' actually excludes everything else

Mate2xo committed:
config(rubocop): ignore policies documentation (obvious)

Mate2xo committed:
config(admin): set default namespace instead of global monkey patch

Mate2xo committed:
chore: lint

Mate2xo committed:
config(tests): require pundit only on policy specs

This avoids having a NoMethodError: undefined method `deprecator' for
module ActiveSupport.
This errors popped when upgrading from Rails 7.0 to Rails 7.1

Mate2xo committed:
Delete unused .coffee files

Mate2xo committed:
build(recurring_select): bump to v4 for rails 7.1 support

Mate2xo committed:
chore: lint

Mate2xo committed:
refactor(layout): better UX, namespace partial to where it's used

Also add missing translations

Mate2xo committed:
chore: lint

Mate2xo committed:
fix(profil icon): broken src link. Use Font-awesome with gem instead

Mate2xo committed:
test: improve coverage

Mate2xo committed:
feat(Rails upgrade): migrate to 7.1

Mate2xo committed:
feat(Rails upgrade): apply 7.1 defaults

Mate2xo committed:
config(rubocop): this file is not useful to document right now

Floury-Fabio commented on db/schema.rb:
```diff
@@ -22,19 +21,19 @@
-    t.index ["author_type", "author_id"], name: "index_active_admin_comments_on_author_type_and_author_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["author_type", "author_id"], name: "index_active_admin_comments_on_author"
```
care the `db:migrate` command generate this diff 
![Screenshot from 2025-05-31 17-25-45](https://github.com/user-attachments/assets/70bea955-9d14-4919-9a4a-4795f3054110)

is the name change intentional?

look up the rails version specified in schema.

Mate2xo replied:
Yes, I started all migrations from a fresh DB just to be sure.
Although doing that revealed that a few more changes were needed (explained in the commit messages)

Mate2xo committed:
build: remove concurrent-ruby fix that was necessary before 7.1

Mate2xo committed:
fix(migrations): needed to bump thredded, then sassc, then puma

`rails db:migrate` would crash on thredded forum migration. The
following changes were needed:
- bump thredded, which needed sassc, which needed sprockets 4
- bump puma to v6, for Rack 3 compatibility, which was needed for the
system tests to launch with Capybara
- update the Member model to declare the enum type through `attribute`,
because when the thredded migrations start, the Member model is then
loaded. Since Rails 7.1, enums must be backed by the `attribute` method
or a database column, and at the time when the migration accesses the
Member model, that column `cash_register_proficiency` does not exist yet
(this raised error is reverted on Rails 7.2)

Mate2xo committed:
fix(thredded sassc): would not compile without optional extend

Mate2xo committed:
chore: lint

Mate2xo committed:
fix(locale): translations were incorrect inside the gem

Mate2xo committed:
test(shoulda-matchers): bump to v6 for Rails7 + Ruby3 compatibility
